pages/doc/integrations_tas_custom_app_metrics_howto.md

```diff
@@ -0,0 +1,75 @@
+
+Pushing metrics can produce more detail; in the push model, a metric can be submitted as often as practically limits allow - possibly multiple times per second. In a scrape model, only changes since the last scrape are seen. Typically metrics data does not need to be extremely high-fidelity in order to be useful, and this periodic sampling is acceptable to most users.
+
+Very short-lived processes **must** use a push model, because they do not run long enough to be discovered and scraped by a metrics collector.
+
+## Metrics Registrar
+
+TAS supports a pull model for custom app metrics using the “Metrics Registrar” component.
+
+The metrics registrar is a TAS component that periodically scrapes registered endpoints and forwards scraped metrics to the Loggregator Firehose. In the Wavefront Nozzle, these metrics are then ingested in the same way as platform or service tile metrics, and become visible in Wavefront automatically.
+
+To instrument an application for metrics registrar scraping, you will typically use either a prometheus SDK or an metrics framework built into your application stack, such as Spring Boot Actuator.
+
+After deploying your application, you register it using the `cf` CLI, as described in the documentation: [https://docs.pivotal.io/application-service/metric-registrar/using.html](https://docs.pivotal.io/application-service/metric-registrar/using.html).
+
+## Wavefront Service Broker
+
+To implement a push-based metrics pattern using Wavefront, you can use the Wavefront SDK for your language stack, or you can use a pluggable abstraction like Spring Boot Actuator.
+
+In either case, you will need to provide your application with a URL for a wavefront proxy instance.
+
+The supported way to do this using the Wavefront Nozzle for TAS is to use the Wavefront Service Broker to create a service instance and bind it to your application. You can then push custom metrics to the proxy from your own application code.
+
+## Factors to Consider
+
+### Spring Boot
+
+Spring Boot is commonly used to build apps running on TAS.
+
+Spring Boot Actuator supports both push and pull models, with the only difference typically being configuration changes. In that case, it becomes relatively easy to switch from push to pull or vice-versa.
+
+### Wavefront Proxy Load
+
+All additional custom app metrics increase the load on the Wavefront Proxy. It may be necessary to increase the count of wavefront proxy instances and to deploy an IaaS load balancer to balance traffic across proxy instances.
+
+### Loggregator Firehose and Nozzle Component Load
+
+The Metrics Registrar scrapes metrics endpoints and publishes those metrics through the Loggregator Firehose. Sending additional app metrics through the Loggregator Firehose may require scaling up the Loggregator Firehose. The Firehose can be horizontally scaled to a certain point, but eventually hits a practical limit where additional instances do not increase throughput.
```

Comment:
I deleted the sentence for future versions -- we don't say what will be or may become possible in tech docs. :)

The Metric Registrar scrapes metrics endpoints and publishes those metrics through the Loggregator Firehose. Sending additional app metrics through the Loggregator Firehose might require scaling up the Loggregator Firehose. The Firehose can be horizontally scaled to a certain point, but eventually hits a practical limit where additional instances do not increase throughput.

The additional metrics flowing through the Firehose also increase the load on two of the Tanzu Observability by Wavefront Nozzle components that cannot be horizontally scaled - the Exporters and the Telegraf Agent. In the current versions, you can increase throughput of these components by increasing the VM size, but only to a certain point. After this point, you must limit the number of metrics or reduce scrape frequency in order to improve performance.

Most TAS environments will not encounter these scaling bottlenecks. If these scaling issues do occur, consider using the push model for sending metrics.